report/4.Solution/DesignOfCompiler.tex

```diff
@@ -0,0 +1,26 @@
+% !TeX root = ../main.tex
+\section{Design of Compiler}
+The compiler is composed of three primary parts, which is explained in the following sections.
+The explanation both covers code snippets of the compiler as well as reasoning behind development decisions.
+However it is also quite import to address the decisions regarding each component on a larger scale, regarding each phase of the compilation process, also to introduce the components of the compiler before they are explained in depth.
```

Comment:
comma after however.
import -> important
after the first "regarding"- delete and truncate up to "each phase"
after "process," replace 'also' with 'and'